techmatters/terraso-mobile-client
# fix: debounce for search queries

## Description

Wrap the map query suggestion system in a 500ms debounce to avoid hammering the map or related downstream APIs. Also refactor the component to extract the query suggestion system to a self-contained hook which the component uses. Additionally, wrap relevant functions in `useCallback` and handle `zoomTo` calls through a single side-effect driven by the suggestion system's selected coordinates rather than scattered through async functions. We believe that the hang issue was caused by some combination of these factors (many async calls affecting zoom position) and that this will mitigate the issue, but were not able to find a direct repro case.

### Checklist

### Related Issues
#1827 

### Verification steps

Enter string and coordinate-based queries in the sites screen search bar and verify that behavior is still as expected, with the addition of a 500ms debounce window before input results in on-screen changes.
